src/diffusers/models/adapter.py

```diff
@@ -0,0 +1,244 @@
+from typing import List, Optional
+
+import torch
+import torch.nn as nn
+
+from ..configuration_utils import ConfigMixin, register_to_config
+from .modeling_utils import ModelMixin
+from .resnet import Downsample2D
+
+
+class BottleneckResnetBlock(nn.Module):
```

Comment:
Let's make sure the naming matches here - we cannot just add any kind of new ResNet all over the place now without thinking about how we can structure them correctly down the road.

Reply:
Note that if we want to later move multiple ResNets together we need to make sure that the name matches